File: back/src/main/java/com/example/intercactivetraining/service/TestService.java
package com.example.intercactivetraining.service;

import com.example.intercactivetraining.model.TestEntity;
import com.example.intercactivetraining.model.TestQuestionAnswerEntity;
import com.example.intercactivetraining.model.UserTestQuestionAnswers;

public interface TestService {
    int saveTest(TestEntity testEntity);

    void saveTestQuestionAnswers(TestQuestionAnswerEntity[] testQuestionAnswers);

    TestEntity getCourseTest(int courseId);

    String checkAnswers(UserTestQuestionAnswers userTestQuestionAnswers, int userId);
}

File: front/interactive-training-platform/src/app/services/test.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {Observable} from "rxjs";
import {CourseTestModel} from "../models/course-test.model";
import {TestQuestionAnswerModel} from "../models/test-question-answer.model";

@Injectable({
  providedIn: 'root'
})
export class TestService {

  private readonly TEST_API: string = "http://localhost:8080/api/test/"

  constructor(private http: HttpClient) { }

  public saveTest(test: CourseTestModel): Observable<string> {
    const options: Object = {
      headers: new HttpHeaders(),
      responseType: 'text'
    }
    const testCopy: any = JSON.parse(JSON.stringify(test));
    testCopy.testContent = JSON.stringify(test.testContent);
    return this.http.post<string>(this.TEST_API + "saveTest", testCopy, options);
  }

  public saveTestQuestionAnswers(testAnswers: TestQuestionAnswerModel[]): Observable<void> {
    return this.http.post<void>(this.TEST_API + "saveTestQuestionAnswers", testAnswers);
  }

  public getCourseTest(courseId: number): Observable<CourseTestModel> {
    return this.http.get<CourseTestModel>(this.TEST_API + "getCourseTest/" + courseId);
  }

}
